sympy/polys/polytools.py

```diff
@@ -5611,10 +5611,18 @@ def _symbolic_factor_list(expr, opt, method):
     """Helper function for :func:`_symbolic_factor`. """
     coeff, factors = S.One, []
 
-    for arg in Mul.make_args(expr):
+    args = list(Mul.make_args(expr))
+    for arg in args:
         if arg.is_Number:
             coeff *= arg
             continue
+        elif hasattr(arg, '_eval_factor'):
+            for i in Mul.make_args(arg._eval_factor()):
```

Comment:
In your version, `base, exp` from the second if statement get overwritten by `arg, S.One`.  Tests fail as a result.  Instead, how about

```python
def _symbolic_factor_list(expr, opt, method):
    """Helper function for :func:`_symbolic_factor`. """
    coeff, factors = S.One, []

    args = list(Mul.make_args(expr))
    for arg in args:
        if arg.is_Number:
            coeff *= arg
            continue
        if hasattr(arg, '_eval_factor'):
            arg = arg._eval_factor()
			if arg.is_Mul:
				args.extend(arg.args)
				continue
        if arg.is_Pow:
            base, exp = arg.args
            if base.is_Number:
                factors.append((base, exp))
                continue
        else:
            base, exp = arg, S.One

        try: ...
```